mailpoet/lib/Form/Block/Text.php

```diff
@@ -34,27 +34,34 @@ public function __construct(
   public function render(array $block, array $formSettings): string {
     $type = 'text';
     $automationId = ' ';
+    $id = '';
     $autocomplete = 'on';
     if ($block['id'] === 'email') {
       $type = 'email';
       $autocomplete = 'email';
     }
 
     if (in_array($block['id'], ['email', 'last_name', 'first_name'], true)) {
+
       $automationId = 'data-automation-id="form_' . $this->wp->escAttr($block['id']) . '" ';
+
+      if (isset($formSettings['id'])) {
+        $id = 'id="form_' . $this->wp->escAttr($block['id']) . '_' . $this->wp->escAttr($formSettings['id']) . '" ';
+      }
     }
```

Comment:
I'm curious why we are restricting **id** and **htmlFor** to only this array ['email', 'last_name', 'first_name'].
What about custom field text input types?

Reply:
@triple0t Good question!
I am not sure of why the restriction is there. It is coming from this [commit](https://github.com/mailpoet/mailpoet/pull/2634/commits/bbfd137a3c2e514df66c774a815981dfbdf2d69b). 
There is another issue in the epic that targets form controls more widely, I have added a note about custom text fields there and specified better the scope of this issue.

Reply:
Hi @brezocordero 
I mean this was added specifically in this commit https://github.com/mailpoet/mailpoet/pull/4171/commits/09ac83cf8efda93997a9ae16f44daf337aa8cdf5#diff-53657c1788edfbe586720f9a5024fd9578cac10853dd2c4602daea4dd2bcdb7aR48-R50
and I was wondering why you scoped the ID to work only under the array 
```
 if (isset($formSettings['id'])) {
        $id = 'id="form_' . $this->wp->escAttr($block['id']) . '_' . $this->wp->escAttr($formSettings['id']) . '" ';
  }
```

I can see [another ticket](https://mailpoet.atlassian.net/browse/MAILPOET-3729) in the backlog addressing this issue so I guess it's fine.

BTW, I can see the same errors report when I test with https://achecker.achecks.ca/checker/index.php (Mostly for forms with custom fields)
<img width="1369" alt="Screenshot 2022-06-03 at 9 49 19 PM" src="https://user-images.githubusercontent.com/30554163/171950939-74f97481-1dc9-4a65-962f-77baebb626ec.png">
<img width="1407" alt="Screenshot 2022-06-03 at 9 49 27 PM" src="https://user-images.githubusercontent.com/30554163/171950964-b5ebefd9-a66f-4de3-b1b6-39f6acbc3b0f.png">

I will add these notes to the ticket and move this PR to QA.

Thank you

Reply:
@triple0t I made the changes and updated the other ticket. Your point made more sense.

Reply:
@brezocordero Thank you for working on it 👏🏾